backend/src/pong/pong.gateway.ts

```diff
@@ -10,117 +11,142 @@ import {
 import { PongService } from './pong.service';
 
+const gameSubscribers: Record<number, GameState> = {};
+
 @WebSocketGateway({
 	namespace: '/pong',
 	cors: {
 		origin: '*',
 	},
 })
 export class PongGateway implements OnGatewayConnection, OnGatewayDisconnect {
-	constructor(private readonly pongService: PongService, private readonly authService: AuthService) {}
-	@WebSocketServer()
-	server: Server;
-
-	private gameState: GameState;
-	private playerOneIsInGame: boolean = false;
-	private playerTwoIsInGame: boolean = false;
-	private playerOneId: number = -1;
-
-
+	constructor(
+		private readonly pongService: PongService,
+		private readonly authService: AuthService,
+	) {}
+	@WebSocketServer() server: Server;
 
-	buildUniqueRoomId(roomInfo: Game): string {
-		return `${roomInfo.id}${Math.min(roomInfo.player_one.id,roomInfo.player_two.id,)}${Math.max(roomInfo.player_one.id, roomInfo.player_two.id)}`;
-		}
-		
 	handleConnection() {
+		this.resetService()
 		console.log('\n!Socket is connected!\n');
-		this.pongService.gameIsFinished = false;
-		this.pongService.pointIsOver = false;
-		this.gameState = this.pongService.createNewGameState();
-		//  set score to win with requestGame info
-		this.gameState.scoreToWin = 10;
 	}
 
 	handleDisconnect(@ConnectedSocket() client: Socket) {
+		const cookie: string = jwtCookieFromHandshakeString(
+			client.handshake.headers.cookie,
+		);
+		const userId: number = this.authService.userIdFromCookieString(cookie);
+			// this.server.in( .roomName).emit('gameOver');
+			// gameSubscribers[data!.id] = undefined;
+			this.pongService.gameIsFinished = true;
 	}
 
-	@SubscribeMessage("joinGameRoom") 
+	createGameInstance(game: Game, client: Socket) {
+		const roomName = String(game.id);
+		gameSubscribers[game.id] = this.pongService.createNewGameState();
+		gameSubscribers[game.id].gameId = game.id;
+		gameSubscribers[game.id].scoreToWin = 10;
+		client.join(roomName);
+		gameSubscribers[game.id].roomName = roomName;
+	}
+
+	@SubscribeMessage('joinGameRoom')
 	joinGameRoom(@MessageBody() game: Game, @ConnectedSocket() client: Socket) {
+		console.log('CurretnGame: ', game);
 		const cookie: string = jwtCookieFromHandshakeString(
 			client.handshake.headers.cookie,
 		);
 		try {
 			const userId: number = this.authService.userIdFromCookieString(cookie);
-			if (userId === game.player_one.id) {
-				this.playerOneIsInGame = true;
-				this.playerOneId = game.player_one.id;
-			}
-			else if (userId === game.player_two.id) {
-				this.playerTwoIsInGame = true;
-			}
-			const roomName = this.buildUniqueRoomId(game);
-			client.join(roomName);
-			this.gameState.roomName = roomName;
-			if (this.playerOneIsInGame && this.playerTwoIsInGame) {
-				this.server.in(roomName).emit("GameCanStart");
+			if (game.id && game.player_one && game.player_two) {
+				if (!gameSubscribers[game.id]) {
+					this.createGameInstance(game, client);
+				} else {
+					client.join(gameSubscribers[game.id].roomName);
+				}
```

Comment:
i think it's neater to not join in the `createGameInstance` method (since it's not really part of creating the instance)
and instead do this:
```ts
if (!gameSubscribers[game.id]) {
    this.createGameInstance(game, client);
}
client.join(gameSubscribers[game.id].roomName);
```

basically make sure that the instance exist and always join afterwards